src/polyfem/State.cpp

```diff
@@ -596,14 +596,8 @@ namespace polyfem
 			{
 				assert(b_discr_orders[i]["id"].is_array() || b_discr_orders[i]["id"].is_number_integer());
 
-				std::vector<int> ids;
-				if (b_discr_orders[i]["id"].is_array())
-					ids = b_discr_orders[i]["id"].get<decltype(ids)>();
-				else
-					ids.push_back(b_discr_orders[i]["id"]);
-
 				const int order = b_discr_orders[i]["order"];
-				for (const int id : ids)
+				for (const int id : json_as_array<int>(b_discr_orders[i]["id"]))
```

Comment:
how this works if `b_discr_orders[i]["id"]` is not an array?

Reply:
This is a util function I wrote that will return a std::vector<int> either directly from the array or with one element.